tello_sim/command_server.py

```
@@ -109,178 +117,200 @@ def listen(self) -> None:
+                    except OSError:
+                        pass
         except KeyboardInterrupt:
             print("\n[Command Listener] Shutting down...")
+        finally:
             self.cleanup()
-        except Exception as e:
-            print(f"[Command Listener] Error: {e}")
-            self.cleanup()
-            raise
+
+    def _handle_connection(self, conn: socket.socket) -> None:
```

Ruff (0.15.20)

[warning] Too many branches (52 > 12)

(PLR0912)